api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/AbstractAccountServiceImpl.java

```diff
@@ -88,55 +97,95 @@ public AccountList getAllAccounts(String columnName, int page, int size, String
-        String search = StringUtils.isEmpty(reqBody.getSearchtext()) ? "" : reqBody.getSearchtext();
         Map<String, String> sortOrderFilter = reqBody.getSortFilter();
         String sortElement = AdminConstants.ACCOUNT_ID;
         String sortOrder = AdminConstants.ASC;
-        if(sortOrderFilter != null &&  sortOrderFilter.containsKey(AdminConstants.SORT_ELEMENT)){
-            sortElement=  sortOrderFilter.get(AdminConstants.SORT_ELEMENT);
-            sortOrder = sortOrderFilter.containsKey(AdminConstants.SORT_ORDER) ? sortOrderFilter.get(AdminConstants.SORT_ORDER) :  AdminConstants.ASC;
+        if (sortOrderFilter != null && sortOrderFilter.containsKey(AdminConstants.SORT_ELEMENT)) {
+            sortElement = sortOrderFilter.get(AdminConstants.SORT_ELEMENT).equalsIgnoreCase("assets") ||
+                    sortOrderFilter.get(AdminConstants.SORT_ELEMENT).equalsIgnoreCase("violations")
+                    ? "cast("+sortOrderFilter.get(AdminConstants.SORT_ELEMENT)+" as decimal(10,2))" : sortOrderFilter.get(AdminConstants.SORT_ELEMENT);
+            sortOrder = sortOrderFilter.containsKey(AdminConstants.SORT_ORDER) ? sortOrderFilter.get(AdminConstants.SORT_ORDER) : AdminConstants.ASC;
         }
-        if(reqBody.getFilter() == null){
-            return convertToMap(accountsRepository.findAll(PageRequest.of(reqBody.getPage(), reqBody.getSize(),
-                            Sort.by(sortOrder.equalsIgnoreCase(AdminConstants.ASC) ? Sort.Direction.ASC : Sort.Direction.DESC, sortElement)),
-                    search.toLowerCase()));
+        String accountQuery = fetchQueryForAccountsFilterApplied(reqBody, sortOrder, sortElement);
+        List<Map<String, Object>> responseList = rdsRepository.getDataFromPacman(accountQuery);
+        List<AccountDetails> accountDetailsList = new ArrayList<>();
+        responseList.stream().forEach(obj -> {
+            AccountDetails accountDetails = new AccountDetails(
+                    (String) (obj.get("accountId") == null ? "N/A" : obj.get("accountId")),
+                    (String) (obj.get("accountName") == null ? "N/A" : obj.get("accountName")),
+                    (String) (obj.get("assets") == null ? "0" : obj.get("assets")),
+                    (String) (obj.get("violations") == null ? "0" : obj.get("violations")),
+                    (String) (obj.get("accountStatus") == null ? "N/A" : obj.get("accountStatus")),
+                    (String) (obj.get("platform") == null ? "N/A" : obj.get("platform")),
+                    (String) (obj.get("createdBy") == null ? "N/A" : obj.get("createdBy")),
+                    (Timestamp) (obj.get("createdTime")));
+            accountDetailsList.add(accountDetails);
+        });
+
+        return convertToMap(new PageImpl<>(accountDetailsList));
+    }
+
+    private String fetchQueryForAccountsFilterApplied(PluginRequestBody reqBody, String sortOrder, String sortElement) {
```

Comment:
This method addresses azure source, however the method name is generic. Can we generalize the sql statement?

Reply:
it is a generic sql statement only. we have cf_Accounts table for all sources, but for azure we have a separate table cf_AzureTenantSubscription, so we need to write case when statement to deal with azure.